src/js/graph/graph.js

```diff
@@ -25,65 +34,100 @@ Graph.prototype.addSchemaUri = function(uri) {
 Graph.prototype.getSchemaUris = function() {
     return this._schemaUris.map(function(schema) { return schema; });
 };
+
 Graph.prototype.setContent = function(content) {
     this._content = content;
 };
 Graph.prototype.getContent = function() {
     return this._content;
 };
+
 Graph.prototype.setContentType = function(contentType) {
     this._contentType = contentType;
 };
 Graph.prototype.getContentType = function() {
     return this._contentType;
 };
+
 Graph.prototype.setNodeTypeAsSpan = function(nodeType, attrMap) {
     this._spanNodeTypes[nodeType] = attrMap || { };
 };
 Graph.prototype.getNodeTypesAsSpan = function() {
     // Don't return a direct reference to the object.
     return deepClone(this._spanNodeTypes);
 };
+
 Graph.prototype.setNodeTypeAsSequence = function(nodeType, attrMap) {
     this._sequenceNodeTypes[nodeType] = attrMap || { };
 };
 Graph.prototype.getNodeTypesAsSequence = function() {
     // Don't return a direct reference to the object.
     return deepClone(this._sequenceNodeTypes);
 };
+
 Graph.prototype.setNodeTypeAsSpanContainer = function(nodeType, attrMap) {
     this._spanContainerNodeTypes[nodeType] = attrMap || { };
 };
 Graph.prototype.getNodeTypesAsSpanContainer = function() {
     // Don't return a direct reference to the object.
     return deepClone(this._spanContainerNodeTypes);
 };
-Graph.prototype.addNode = function(node, connectEdges) {
+
+// Node manipulation functions
+Graph.prototype.getNodeIds = function() {
+    return this._graphImpl.nodes();
+};
+Graph.prototype.getNodes = function() {
+    return this.getNodeIds().map(function(nodeId) {
+        return this.getNodeById(nodeId);
+    }, this);
+};
+Graph.prototype.nodeExists = function(nodeId) { return this._graphImpl.hasNode(nodeId); };
+Graph.prototype.getNodeById = function(nodeId) { return this._graphImpl.node(nodeId); };
+Graph.prototype.getNodeTypes = function() { return Object.keys(this._nodeTypes); };
+Graph.prototype.getNodesByType = function(nodeType) {
+    if (this._nodeTypes[nodeType] === undefined) { return []; }
+    return this._nodeTypes[nodeType].map(function(node) {
+        return node;
+    });
+};
+Graph.prototype.getNodeIds = function() {
```

Comment:
More leftovers from merging the experimental compound graph branch into the dev branch. Deleted